trulens_eval/trulens_eval/keys.py

```diff
@@ -76,16 +155,39 @@ def get_huggingface_headers():
     }
     return HUGGINGFACE_HEADERS
 
+def _check_key(k, v = None):
+    v = v or os.environ.get(k)
 
-def setup_keys(**kwargs):
-    global config_file
+    if v is None or "fill" in v:
+        raise RuntimeError(\
+f"""{UNICODE_STOP} Variable {k} needs to be set; please provide it in one of these ways:
+
+  - in a variable {k} prior to this check, 
+  - in your variable environment, 
+  - in a .env file in {Path.cwd()} or its parents,
+  - explicitly passed to function `check_keys` of `trulens_eval.keys`,
+  - passed to the endpoint or feedback collection constructor that needs it (`trulens_eval.provider_apis.OpenAIEndpoint`, etc.), or
+  - set in api utility class that expects it (i.e. `openai`, etc.).
+
+For the last two options, the name of the argument may differ from {k} (i.e. `openai.api_key` for `OPENAI_API_KEY`).
+""")
+
+
+def check_keys(**kwargs):
```

Comment:
This call is just confusing to me... It adds one layer of API call on top of the already used set key methods. Then, if they are already set; its going to ignore what you type in here? But then theres no way to simply check the key without adding bogus string in.

_check_key makes sense because if no val is supplied it, will check existence. Can we just remove the "set" functionality of this method?

Reply:
Did this. The setting functionality is only part of new `check_or_set_keys`.